monkey/tests/unit_tests/common/agent_configuration/test_agent_configuration.py

```diff
@@ -186,70 +152,60 @@ def test_network_scan_configuration():
 
 def test_exploitation_options_configuration_schema():
     ports = [1, 2, 3]
-    schema = ExploitationOptionsConfigurationSchema()
 
-    config = schema.load({"http_ports": ports})
+    config = ExploitationOptionsConfiguration(**{"http_ports": ports})
 
     assert config.http_ports == tuple(ports)
 
 
 @pytest.mark.parametrize("ports", INVALID_PORTS)
 def test_exploitation_options_configuration_schema__ports_out_of_range(ports):
-    schema = ExploitationOptionsConfigurationSchema()
-
     invalid_ports_configuration = {"http_ports": ports}
 
-    with pytest.raises(ValidationError):
-        schema.load(invalid_ports_configuration)
+    with pytest.raises(ValueError):
+        ExploitationOptionsConfiguration(**invalid_ports_configuration)
 
 
 def test_exploiter_configuration_schema():
     name = "bond"
     options = {"gun": "Walther PPK", "car": "Aston Martin DB5"}
-    schema = PluginConfigurationSchema()
 
-    config = schema.load({"name": name, "options": options})
+    config = PluginConfiguration(**{"name": name, "options": options})
 
     assert config.name == name
     assert config.options == options
 
 
 def test_exploitation_configuration():
-    schema = ExploitationConfigurationSchema()
-
-    config = schema.load(EXPLOITATION_CONFIGURATION)
-    config_dict = schema.dump(config)
+    config = ExploitationConfiguration(**EXPLOITATION_CONFIGURATION)
+    config_dict = config.dict()
 
     assert isinstance(config, ExploitationConfiguration)
-    assert config_dict == EXPLOITATION_CONFIGURATION
+    assert config_dict == convert_lists_to_tuples(EXPLOITATION_CONFIGURATION.copy())
```

Comment:
Why is `EXPLOITATION_CONFIGURATION` not using tuples?

Reply:
We could modify the configurations but I figured it's worth keeping them as is and making sure lists are accepted during configuration construction.